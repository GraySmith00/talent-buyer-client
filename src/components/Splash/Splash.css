.splash {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.5rem;
  width: 60%;
  text-align: justify;
  background-color: rgba(0, 0, 0, 0.8);
  margin: 30px 0 100px 0;
  border-radius: 10px;
  color: #fff; }
  .splash p {
    margin: 0; }
  .splash .buttons {
    display: flex;
    justify-content: space-around;
    margin-top: 2rem;
    grid-gap: 20px;
    width: 100%; }
