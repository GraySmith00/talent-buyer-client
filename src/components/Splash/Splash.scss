.splash {
  display: flex;
  flex-direction: column;
  align-items: center;

  padding: 1.5rem;
  width: 60%;
  text-align: justify;

  background-color: rgba(0, 0, 0, 0.8);
  margin: 30px 0 100px 0;
  border-radius: 10px;

  color: #fff;

  p {
    margin: 0;
  }

  .buttons {
    display: flex;
    justify-content: space-around;
    grid-gap: 20px;
    width: 100%;
  }
}
