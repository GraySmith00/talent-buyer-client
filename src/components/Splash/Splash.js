import React, { Component } from 'react';
import ModalButton from '../styledComponents/ModalButton';
import PropTypes from 'prop-types';
import './Splash.css';

export default class Splash extends Component {
  render() {
    return (
      <div className="splash">
        <div className="button-container">
          <p>
            Marfa tousled freegan, health goth everyday carry prism four dollar
            toast raclette blog. Pitchfork waistcoat +1 mlkshk roof party
            aesthetic humblebrag ramps, yr selvage. Trust fund echo park
            sartorial put a bird on it. Twee ennui bushwick celiac try-hard
            pinterest. Bushwick microdosing edison bulb banh mi poutine DIY.
            Cornhole tote bag twee, bespoke ramps roof party
          </p>

          <ModalButton
            className="half-button"
            onClick={this.props.openSignUpModal}
          >
            Sign Up
          </ModalButton>
          <ModalButton
            className="half-button"
            onClick={this.props.openLogInModal}
          >
            Log In
          </ModalButton>
        </div>
      </div>
    );
  }
}

Splash.propTypes = {
  openSignUpModal: PropTypes.func.isRequired,
  openLogInModal: PropTypes.func.isRequired
};
