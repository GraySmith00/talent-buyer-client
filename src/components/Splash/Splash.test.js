import React from 'react';
import { shallow } from 'enzyme';

import { Splash } from './Splash';

describe('Splash component', () => {
  let wrapper;
  let mockOpenSignUpModal;
  let mockOpenLogInModal;

  beforeEach(() => {
    mockOpenSignUpModal = jest.fn();
    mockOpenLogInModal = jest.fn();

    wrapper = shallow(
      <Splash
        openSignUpModal={mockOpenSignUpModal}
        openLogInModal={mockOpenLogInModal}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
