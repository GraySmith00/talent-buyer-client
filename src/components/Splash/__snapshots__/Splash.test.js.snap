// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    openLogInModal={[Function]}
    openSignUpModal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Talent Buyer is the fastest and easist way for music talent buyers to browse artists and filter by agency, popularity, and genre. Artists can be saved to a watchlist for quick reference later. Buyers can easily build, edit, and download offers and keep track of the calendar for their venues.
</p>,
        <section
          className="buttons"
>
          <Unknown
                    onClick={[Function]}
          >
                    Sign Up
          </Unknown>
          <Unknown
                    onClick={[Function]}
          >
                    Log In
          </Unknown>
</section>,
      ],
      "className": "splash",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Talent Buyer is the fastest and easist way for music talent buyers to browse artists and filter by agency, popularity, and genre. Artists can be saved to a watchlist for quick reference later. Buyers can easily build, edit, and download offers and keep track of the calendar for their venues.",
        },
        "ref": null,
        "rendered": "Talent Buyer is the fastest and easist way for music talent buyers to browse artists and filter by agency, popularity, and genre. Artists can be saved to a watchlist for quick reference later. Buyers can easily build, edit, and download offers and keep track of the calendar for their venues.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              onClick={[Function]}
>
              Sign Up
</Unknown>,
            <Unknown
              onClick={[Function]}
>
              Log In
</Unknown>,
          ],
          "className": "buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Sign Up",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "rules": Array [
                  "
  width: 40%;
  padding: 10px 20px;
  margin-top: 2rem;
  border: none;
  border-radius: 5px;
  font-size: 1.2rem;
  background-color: #d52f7f;
  color: #fff;
  cursor: pointer;
  -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
",
                ],
              },
              "displayName": "styled.button",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Log In",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Log In",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "rules": Array [
                  "
  width: 40%;
  padding: 10px 20px;
  margin-top: 2rem;
  border: none;
  border-radius: 5px;
  font-size: 1.2rem;
  background-color: #d52f7f;
  color: #fff;
  cursor: pointer;
  -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
",
                ],
              },
              "displayName": "styled.button",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "button",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Talent Buyer is the fastest and easist way for music talent buyers to browse artists and filter by agency, popularity, and genre. Artists can be saved to a watchlist for quick reference later. Buyers can easily build, edit, and download offers and keep track of the calendar for their venues.
</p>,
          <section
            className="buttons"
>
            <Unknown
                        onClick={[Function]}
            >
                        Sign Up
            </Unknown>
            <Unknown
                        onClick={[Function]}
            >
                        Log In
            </Unknown>
</section>,
        ],
        "className": "splash",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Talent Buyer is the fastest and easist way for music talent buyers to browse artists and filter by agency, popularity, and genre. Artists can be saved to a watchlist for quick reference later. Buyers can easily build, edit, and download offers and keep track of the calendar for their venues.",
          },
          "ref": null,
          "rendered": "Talent Buyer is the fastest and easist way for music talent buyers to browse artists and filter by agency, popularity, and genre. Artists can be saved to a watchlist for quick reference later. Buyers can easily build, edit, and download offers and keep track of the calendar for their venues.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                onClick={[Function]}
>
                Sign Up
</Unknown>,
              <Unknown
                onClick={[Function]}
>
                Log In
</Unknown>,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Sign Up",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "rules": Array [
                    "
  width: 40%;
  padding: 10px 20px;
  margin-top: 2rem;
  border: none;
  border-radius: 5px;
  font-size: 1.2rem;
  background-color: #d52f7f;
  color: #fff;
  cursor: pointer;
  -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
",
                  ],
                },
                "displayName": "styled.button",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Log In",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log In",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "rules": Array [
                    "
  width: 40%;
  padding: 10px 20px;
  margin-top: 2rem;
  border: none;
  border-radius: 5px;
  font-size: 1.2rem;
  background-color: #d52f7f;
  color: #fff;
  cursor: pointer;
  -webkit-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 57px -16px rgba(0, 0, 0, 0.75);
",
                  ],
                },
                "displayName": "styled.button",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
